src/handler.cpp

```diff
@@ -852,8 +825,33 @@ void affect_total(struct char_data * ch)
 
   apply_drug_modifiers_to_ch(ch);
 
-  skill_dice = get_skill_dice_in_use_for_weapons(ch);
+  // Min attribute is one, max is soft capped
+  cap = ((ch_is_npc || (GET_LEVEL(ch) >= LVL_ADMIN)) ? 50 : 20);
+  for (int att = BOD; att <= WIL; att++) {
+    GET_ATT(ch, att) = MAX(1, GET_ATT(ch, att));
+    if (GET_ATT(ch, att) > cap)
+      GET_ATT(ch, att) = cap + ((GET_ATT(ch, att) - cap + 1) >> 1);
+  }
+  GET_MAG(ch) = MAX(0, MIN(GET_MAG(ch), cap * 100));
+  GET_ESS(ch) = MAX(0, MIN(GET_ESS(ch), 600));
+  GET_MAG(ch) -= MIN(GET_MAG(ch), GET_TEMP_MAGIC_LOSS(ch) * 100);
+  GET_ESS(ch) -= GET_TEMP_ESSLOSS(ch);
+  GET_MAX_MENTAL(ch) = 1000;
+  GET_MAX_PHYSICAL(ch) = 1000;
+  GET_TARGET_MOD(ch) = 0;
+  GET_CONCENTRATION_TARGET_MOD(ch) = 0;
+  GET_MAX_MENTAL(ch) -= GET_MENTAL_LOSS(ch) * 100;
+  GET_MAX_PHYSICAL(ch) -= GET_PHYSICAL_LOSS(ch) * 100;
+
+  // Here so that other modifiers can't make the char immune to disabling via nerve strike
+  if (GET_TEMP_QUI_LOSS(ch))
+    GET_QUI(ch) = MAX(0, GET_QUI(ch) - (GET_TEMP_QUI_LOSS(ch) / TEMP_QUI_LOSS_DIVISOR));
 
+  // Reaction is derived from current atts, so we calculate it after all att modifiers
+  // We don't cap reaction because qui and int are already capped
+  GET_REA(ch) += (GET_INT(ch) + GET_QUI(ch)) >> 1;
```

Comment:
Did GET_REA() get zeroed before this? If not, this will cause it to explode.

Reply:
Yeah, that happens on line 581, which surprisingly is still in that same function despite being 300 lines earlier >.<

Reply:
It looks like where GET_REA is zeroes is also where all the attributes are reset to their real values and the pools are emptied. After all that, we start adding everything back in, including bonuses to reaction directly from things like wired reflexes and spells. Unless we want to tackle breaking this function down into more digestible subfunctions, where GET_REA is currently zeroed does make sense.